spec/api-browser-window-spec.ts

```diff
@@ -6983,4 +6983,331 @@ describe('BrowserWindow module', () => {
+          expect(savedState.right).to.be.lessThanOrEqual(savedState.work_area_right);
+          expect(savedState.bottom).to.be.lessThanOrEqual(savedState.work_area_bottom);
+        });
+
+        it('should save work area bounds that contain the window bounds on secondary display', async function () {
+          // Fixture will center the window on any secondary display if available
+          const appPath = path.join(fixturesPath, 'work-area-secondary');
+          const appProcess = childProcess.spawn(process.execPath, [appPath]);
+          const [code] = await once(appProcess, 'exit');
+
+          // Fixture returns code 1 due to single monitor setup
+          if (code === 1) {
+            console.log('Skipping secondary display test - only one monitor available');
+            this.skip();
+            return;
+          }
+
+          expect(code).to.equal(0);
+
+          const savedState = getWindowStateFromDisk('test-work-area-secondary', sharedPreferencesPath);
+          expect(savedState).to.not.be.null('window state with id "test-work-area-secondary" does not exist');
+
+          expect(savedState.left).to.be.greaterThanOrEqual(savedState.work_area_left);
+          expect(savedState.top).to.be.greaterThanOrEqual(savedState.work_area_top);
+          expect(savedState.right).to.be.lessThanOrEqual(savedState.work_area_right);
+          expect(savedState.bottom).to.be.lessThanOrEqual(savedState.work_area_bottom);
+        });
+      });
+
+      describe('asynchronous batching behavior', () => {
+        let w: BrowserWindow;
+        const stateId = 'test-batching-behavior';
+        const preferencesPath = path.join(app.getPath('userData'), 'Preferences');
+
+        // Helper to get preferences file modification time
+        const getPrefsModTime = (): Date => {
+          try {
+            return fs.statSync(preferencesPath).mtime;
+          } catch {
+            throw new Error(`Test requires preferences file to exist at path: ${preferencesPath}.`);
+          }
+        };
+
+        // Helper to wait for file modification with 20 second default timeout
+        const waitForPrefsUpdate = async (initialModTime: Date, timeoutMs: number = 20000): Promise<void> => {
+          const startTime = Date.now();
+
+          while (true) {
+            const currentModTime = getPrefsModTime();
+
+            if (currentModTime > initialModTime) {
+              return;
+            }
+
+            if (Date.now() - startTime > timeoutMs) {
+              throw new Error(`Window state was not flushed to disk within ${timeoutMs}ms`);
+            }
+            // Wait for 1 second before checking again
+            await setTimeout(1000);
+          }
+        };
+
+        beforeEach(() => {
+          w = new BrowserWindow({
+            show: false,
+            width: 400,
+            height: 300,
+            // @ts-ignore: remove this once the type definition is added
```

Comment:
```suggestion
            // @ts-expect-error: remove this once the type definition is added
```

If we use `@ts-expect-error` instead of `@ts-ignore`, it should start complaining once the definition is added so we are forced to clean this up.